n2v/methods/wuyang.py

```diff
@@ -129,18 +152,135 @@ def hessian(self, v):
         Ha = 2 * contract('iau,iat,ia->ut', C3a, C3a, eigs_diff_a**-1)
 
         if self. ref == 1:
+            if self.lambda_reg is not None:
+                Ha += 4 * self.T_pbs
             Hs = Ha
 
         else:
             eigs_diff_b = self.eigvecs_b[:nb, None] - self.eigvecs_b[None, nb:]
             C3b = contract('mi,va,mvt->iat', self.Cb[:,:nb], self.Cb[:,nb:], self.S3)
             Hb = 2 * contract('iau,iat,ia->ut', C3b, C3b, eigs_diff_b**-1)
-            Hs = np.block( 
-                            [[Ha,                               np.zeros((self.naux, self.naux))],
-                            [np.zeros((self.naux, self.naux)), Hb                              ]] 
+            if self.lambda_reg is not None:
+                Hb += 4 * self.T_pbs
+            Hs = np.block(
+                            [[Ha,                               np.zeros((self.npbs, self.npbs))],
+                            [np.zeros((self.npbs, self.npbs)), Hb                              ]]
                         )
 
-        if self.reg > 0.0:
-            pass
-
         return - Hs
+
+    def find_regularization_constant(self, opt_method, potential_components, tol=None, opt=None, lambda_list=None):
```

Comment:
So is this the same as running WuYang?. You first do one optimization, and with the result, you do it again with many regularization constants? And then stick with the one that works best?

Reply:
Yes. I didn't realized you it was committed here. There is actually minor problems in the functions that I fixed, which I don't think is committed yet.